src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2229,12 +2230,12 @@ private IEnumerable<PENamedTypeSymbol> CreateNestedTypes()
                         if (marker.HasSpecialName && marker.IsStatic && marker.DeclaredAccessibility == Accessibility.Private &&
                             marker.TypeKind == TypeKind.Class && marker.BaseTypeNoUseSiteDiagnostics.IsObjectType() && marker.Arity == 0 &&
                             marker.InterfacesNoUseSiteDiagnostics().IsEmpty &&
-                            (type.Arity == 0 || (marker is PENamedTypeSymbolGeneric genericMarker && genericMarker.MatchesContainingTypeParameters())))
+                            (type.Arity == 0 || marker is PENamedTypeSymbolGeneric))
                         {
                             // Try to locate the marker method.
                             if (marker.TryGetExtensionMarkerMethod() is { IsNil: false } markerHandle)
                             {
-                                marker = PENamedTypeSymbol.Create(moduleSymbol, this, markerRid, type.Handle, markerHandle);
+                                marker = PENamedTypeSymbol.Create(moduleSymbol, this, markerRid, marker, markerHandle);
```

Comment:
nit: it didn't catch my eye in earlier PR, but it's strange to assign this to `marker`. A fresh local, maybe named "extension", would be better. I'm happy to do this tweak later on (just let me know if okay)